File: backend/src/main/java/ru/sportexpress/menuApp/web/controller/HomeController.java
package ru.sportexpress.menuApp.web.controller;

import graphql.ExecutionResult;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import ru.sportexpress.menuApp.entity.Menu;
import ru.sportexpress.menuApp.web.service.MenuService;
import ru.sportexpress.menuApp.web.service.GraphQlService;

import java.util.List;
import java.util.Optional;

@RestController
@CrossOrigin
public class HomeController {

    @Autowired
    private MenuService menuService;

    @Autowired
    GraphQlService graphQLService;

    @GetMapping(value = "")
    public List<Menu> getAll() {
        return menuService.getAll();
    }

    @GetMapping(value = "/{id}")
    public Optional<Menu> getCustomer(@PathVariable("id") int id) {
        return menuService.getOne(id);
    }

    @PostMapping(value = "")
    public Menu saveCustomer(@RequestBody Menu customer) {
        return menuService.save(customer);
    }

    @PutMapping(value = "/{id}")
    public Menu updateCustomer(@PathVariable(name = "id") int id, @RequestBody Menu customer) {
        return menuService.update(id, customer);
    }

    @DeleteMapping(value = "/{id}")
    public Menu deleteCustomer(@PathVariable(name = "id") int id) {
        return menuService.delete(id);
    }

    @PostMapping(value = "/graphql")
    public ResponseEntity<Object> getAllCustomers(@RequestBody String query) {
        ExecutionResult execute = graphQLService.getGraphQL().execute(query);
        return new ResponseEntity<>(execute, HttpStatus.OK);
    }
}

File: backend/src/main/kotlin/ru/sportexpress/menuApp/web/controller/HomeController.kt
package ru.sportexpress.menuApp.web.controller

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*
import ru.sportexpress.menuApp.entity.Menu
import ru.sportexpress.menuApp.requests.MenuResponse
import ru.sportexpress.menuApp.web.service.GraphQlService
import ru.sportexpress.menuApp.web.service.MenuService
import java.util.*
import javax.validation.Valid


@RestController
@CrossOrigin
public class HomeController {

    @Autowired
    lateinit var menuService: MenuService 

    @Autowired
    lateinit var graphQLService :GraphQlService

    @GetMapping("/")
    fun getAll() : MenuResponse {
        return menuService.getAll()
    }

    @GetMapping("/{id}")
    fun getCustomer(@PathVariable("id") id : Long) : Optional<Menu> {
        return menuService.getOne(id)
    }

    @PostMapping("")
    fun saveCustomer(@Valid @RequestBody menu: Menu) : Menu{
        return menuService.save(menu)
    }

    @PutMapping("/{id}")
    fun updateCustomer(@PathVariable(name = "id") id :Long, @Valid @RequestBody customer:Menu):Menu {
        return menuService.update(id, customer)
    }

    @DeleteMapping("/{id}")
    fun deleteCustomer(@PathVariable(name = "id") id :Long):Menu {
        return menuService.delete(id)
    }

//    @GetMapping("/swithParent")
//    fun changeParent(@PathParam("data") params : Map<String, Long> ):Menu{
//        return null
//    }
//
//    @PostMapping("/graphql")
//    fun getAllCustomers(@RequestBody query : String) :ResponseEntity<Objects> {
//        val execute : ExecutionResult = graphQLService.getGraphQL().execute(query)
//        return ResponseEntity<>(execute, HttpStatus.OK)
//    }
}
